os/include/signal.h

```diff
@@ -579,6 +579,9 @@ CODE void (*signal(int sig, CODE void (*func)(int sig)))(int sig);
  * @}
  */
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshadow"
+
```

Comment:
If we are using this approach, then pls add comments indicating why we are doing this.

Reply:
Done